Importamos la BD, despues de la importacion tenemos 8012 filas y 25 columnas

In [1]:
import pandas as pd
archivo = '..\data\ISBSG - Release May 2017 R1.csv'
#archivo = '..\data\ISBSG DATA Release 12.csv'
df = pd.read_csv(archivo, sep = ';')
variables = ['ISBSG Project ID', 'Data Quality Rating', 'UFP rating', 'Application Group', 'Development Type', 'Development Platform', 'Language Type', 'Primary Programming Language', 'Count Approach', 'Functional Size', 'Adjusted Function Points', 'Normalised Work Effort Level 1', 'Summary Work Effort', 'Project Elapsed Time', 'Business Area Type', 'Used Methodology', 'Resource Level', 'Max Team Size', 'Average Team Size', 'Input count', 'Output count', 'Enquiry count', 'File count', 'Interface count', 'Agile Method Used']
df = df.loc[:, variables]
df

ISBSG Project ID Data Quality Rating UFP rating     Application Group  \
0                10001                   D          A  Business Application   
1                10003                   B          B  Business Application   
2                10011                   B          A  Business Application   
3                10012                   B          A  Business Application   
4                10014                   B          A                   NaN   
...                ...                 ...        ...                   ...   
8007             34870                   B          B  Business Application   
8008             34901                   B          B                   NaN   
8009             34902                   B          B  Business Application   
8010             34990                   B          B                   NaN   
8011             34997                   B          B                   NaN   

     Development Type Development Platform Language Type  \
0     New Development                   MR           4GL   
1         Enhancement                Multi           NaN   
2     New Development                Multi           4GL   
3         Enhancement                  NaN           3GL   
4         Enhancement                  NaN           NaN   
...               ...                  ...           ...   
8007      Enhancement                  NaN           NaN   
8008  New development                  NaN           4GL   
8009      Enhancement                  NaN           NaN   
8010      Enhancement                  NaN           NaN   
8011  New Development                  NaN           3GL   

     Primary Programming Language Count Approach  Functional Size  ...  \
0                          Oracle          NESMA            237.0  ...   
1                             NaN       IFPUG 4+             67.0  ...   
2                          Access       IFPUG 4+            443.0  ...   
3                           COBOL       IFPUG 4+             76.0  ...   
4                             NaN       IFPUG 4+              3.0  ...   
...                           ...            ...              ...  ...   
8007                          NaN       IFPUG 4+            161.0  ...   
8008                         .Net          NESMA           5667.0  ...   
8009                          NaN       IFPUG 4+             84.0  ...   
8010                       Cordys          IFPUG            184.0  ...   
8011                         Java          IFPUG             61.0  ...   

     Used Methodology  Resource Level  Max Team Size Average Team Size  \
0                  No             1.0              5                 5   
1                 NaN             1.0            NaN               NaN   
2          Don't Know             1.0            NaN               NaN   
3                 Yes             1.0            NaN               NaN   
4                 NaN             1.0            NaN               NaN   
...               ...             ...            ...               ...   
8007              Yes             1.0            NaN               NaN   
8008              Yes             1.0            NaN               NaN   
8009              Yes             1.0            NaN               NaN   
8010              Yes             1.0            NaN               NaN   
8011              Yes             1.0            NaN               NaN   

     Input count Output count  Enquiry count File count Interface count  \
0             60          110           10.0         42            15.0   
1            NaN          NaN            NaN        NaN             NaN   
2            NaN          NaN            NaN        NaN             NaN   
3            NaN          NaN            NaN        NaN             NaN   
4            NaN          NaN            NaN        NaN             NaN   
...          ...          ...            ...        ...             ...   
8007         NaN          NaN    

In [2]:
filtro = ((df['Data Quality Rating'] == 'A') | (df['Data Quality Rating'] == 'B')) & ((df['UFP rating'] == 'A') | (df['UFP rating'] == 'B')) 
df = df.loc[filtro, :]
filtro = (df['Normalised Work Effort Level 1'].notnull()) & (df['Normalised Work Effort Level 1'] == df['Summary Work Effort'])
df = df.loc[filtro, :]
filtro = df['Count Approach'] == 'IFPUG 4+'
df = df.loc[filtro, :]
variables = ['ISBSG Project ID', 'Application Group', 'Development Type', 'Development Platform', 'Language Type', 'Primary Programming Language', 'Functional Size', 'Adjusted Function Points', 'Normalised Work Effort Level 1', 'Project Elapsed Time', 'Business Area Type', 'Used Methodology', 'Resource Level', 'Max Team Size', 'Average Team Size', 'Input count', 'Output count', 'Enquiry count', 'File count', 'Interface count', 'Agile Method Used']
df = df.loc[:, variables]
df = df.dropna(axis=1, thresh=int(0.6*len(df)))
df.dropna()

ISBSG Project ID     Application Group Development Type  \
7                10026  Business Application  New Development   
43               10143  Business Application  New Development   
54               10173  Business Application      Enhancement   
63               10200  Business Application      Enhancement   
68               10218  Business Application  New Development   
...                ...                   ...              ...   
7924             32708  Business Application      Enhancement   
7925             32710  Business Application  New Development   
7928             32721  Business Application      Enhancement   
7934             32731  Business Application  New Development   
7940             32756  Business Application      Enhancement   

     Development Platform Language Type Primary Programming Language  \
7                      MF           3GL                         Java   
43                     MF           3GL                        COBOL   
54                  Multi           3GL                         Java   
63                  Multi           3GL                         Java   
68                     MF           3GL                         PL/I   
...                   ...           ...                          ...   
7924                   MR           3GL                        COBOL   
7925                Multi           3GL                         Java   
7928                   MF           3GL                         PL/I   
7934                Multi           4GL                    Datastage   
7940                   MF           3GL                        COBOL   

      Functional Size Adjusted Function Points  \
7               620.0                      620   
43              243.0                      250   
54               25.0                       25   
63               90.0                       90   
68              163.0                      152   
...               ...                      ...   
7924            242.0                      213   
7925            133.0                      133   
7928             96.0                       96   
7934           1048.0                     1048   
7940             23.0                       23   

      Normalised Work Effort Level 1 Project Elapsed Time Used Methodology  \
7                            18160.0                  7,0       Don't Know   
43                            2924.0                  8,0              Yes   
54                             118.0                  1,0              Yes   
63                            1112.0                  4,0              Yes   
68                             668.0                  9,0              Yes   
...                              ...                  ...              ...   
7924                          2185.0                  7,0               No   
7925                          1769.0                  6,0              Yes   
7928                          1011.0                  6,0              Yes   
7934                          7722.0                 16,0              Yes   
7940                           224.0                 12,5              Yes   

      Resource Level  
7                1.0  
43               1.0  
54               1.0  
63               1.0  
68               1.0  
...              ...  
7924             1.0  
7925             1.0  
7928             1.0  
7934             1.0  
7940             1.0  

[860 rows x 12 columns]

Tras hacer el filtrado de las variables que nos interesa nos quedan 860 proyectos válidos.

Si aplicamos los filtros anteriores, sin desechar variables, y además seleccionamos aquellos que han utilizado metodología agil nos quedan solamente 21 proyectos.

In [2]:
filtro = ((df['Data Quality Rating'] == 'A') | (df['Data Quality Rating'] == 'B')) & ((df['UFP rating'] == 'A') | (df['UFP rating'] == 'B')) 
df = df.loc[filtro, :]
filtro = (df['Normalised Work Effort Level 1'].notnull()) & (df['Normalised Work Effort Level 1'] == df['Summary Work Effort'])
df = df.loc[filtro, :]
filtro = df['Count Approach'] == 'IFPUG 4+'
df = df.loc[filtro, :]
variables = ['ISBSG Project ID', 'Application Group', 'Development Type', 'Development Platform', 'Language Type', 'Primary Programming Language', 'Functional Size', 'Adjusted Function Points', 'Normalised Work Effort Level 1', 'Project Elapsed Time', 'Business Area Type', 'Used Methodology', 'Resource Level', 'Max Team Size', 'Average Team Size', 'Input count', 'Output count', 'Enquiry count', 'File count', 'Interface count', 'Agile Method Used']
df = df.loc[:, variables]
filtro = df['Agile Method Used'] == 'Yes'
df = df.loc[filtro, :]
df

ISBSG Project ID     Application Group Development Type  \
451              11278  Business Application      Enhancement   
629              11801                   NaN      Enhancement   
1505             14319  Business Application  New Development   
1507             14325  Business Application  New Development   
1573             14525  Business Application      Enhancement   
1746             15011  Business Application  New Development   
2095             16039  Business Application      Enhancement   
2510             17275  Business Application  New Development   
2598             17548  Business Application  New Development   
2808             18119                   NaN      Enhancement   
4331             22347  Business Application  New Development   
4756             23552  Business Application  New Development   
5402             25415  Business Application  New Development   
5614             26034  Business Application  New Development   
5671             26195  Business Application      Enhancement   
5819             26585                   NaN  New Development   
5935             26861  Business Application  New Development   
6083             27281                   NaN  New Development   
7087             30367  Business Application      Enhancement   
7791             32296  Business Application  New Development   
7921             32692  Business Application  New Development   

     Development Platform Language Type Primary Programming Language  \
451                 Multi           3GL                   JavaScript   
629                 Multi           3GL                           C#   
1505          Proprietary           5GL   Proprietary Agile Platform   
1507                   MR           4GL                         .Net   
1573                Multi           3GL                           C#   
1746          Proprietary           5GL   Proprietary Agile Platform   
2095                Multi           3GL                           C#   
2510          Proprietary           5GL   Proprietary Agile Platform   
2598          Proprietary           5GL   Proprietary Agile Platform   
2808                Multi           3GL                           C#   
4331                Multi           3GL                   JavaScript   
4756          Proprietary           5GL   Proprietary Agile Platform   
5402          Proprietary           5GL   Proprietary Agile Platform   
5614          Proprietary           5GL   Proprietary Agile Platform   
5671                Multi           4GL                         ABAP   
5819          Proprietary           5GL   Proprietary Agile Platform   
5935          Proprietary           5GL   Proprietary Agile Platform   
6083          Proprietary           5GL   Proprietary Agile Platform   
7087                Multi           3GL                         PL/I   
7791          Proprietary           5GL   Proprietary Agile Platform   
7921          Proprietary           5GL   Proprietary Agile Platform   

      Functional Size Adjusted Function Points  \
451              89.0                       89   
629              63.0                       63   
1505            748.0                      748   
1507            498.0                      498   
1573            695.0                      695   
1746            257.0                      257   
2095             47.0                       47   
2510            168.0                      168   
2598            116.0                      116   
2808            109.0                      109   
4331            399.0                      399   
4756            243.0                      243   
5402            679.0                      679   
5614            143.0                      143   
5671            211.0                      211   
5819            243.0                      243   
5935            138.0                      138   
6083            144.0                      144   
7087            117.0                   